cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -34,3 +39,148 @@ def test_cluster_single_db(cluster_with_single_db, logger, attributes, cfy):
                      cert_path=mgr2.local_ca)
 
     check_managers(mgr1, mgr2)
+
+
+def test_queue_node_failover(cluster_with_single_db, logger,
+                             module_tmpdir, attributes, ssh_key, cfy):
+    hello_world = prepare_cluster_with_agent(
+        cluster_with_single_db, logger,
+        module_tmpdir, attributes, ssh_key, cfy
+    )
+    broker1, broker2, broker3, db, mgr1, mgr2 = cluster_with_single_db
+    validate_cluster_and_agents(mgr1)
+    agent_broker_ip = verify_agent_broker_connection_and_get_broker_ip(mgr1)
+
+    # stop the rabbitmq service in the agent's broker node
+    agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == agent_broker_ip:
+            agent_broker = broker
+            break
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq stop')
+
+    # the agent should now pick another broker
+    validate_cluster_and_agents(mgr1, expected_broker_status='Degraded')
+    new_agent_broker_ip = \
+        verify_agent_broker_connection_and_get_broker_ip(mgr1)
+    assert new_agent_broker_ip != agent_broker_ip
+
+    new_agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == new_agent_broker_ip:
+            new_agent_broker = broker
+            break
+
+    # spin up 3 deployments
```

Comment:
Why are you spinning up three deployments here? The comment says what you're doing but not why.

Reply:
I wanted to do some load-testing on the queues, similar to what Ahmad did. I'm not sure if it's good though so if you have better ideas...